src/Futurefuel/Saml2/Saml2Auth.php

```diff
@@ -204,10 +204,36 @@ function getMetadata()
         $errors = $settings->validateMetadata($metadata);
 
         if (empty($errors)) {
+            $spData = $settings->getSPData();
+            $attributeServiceXml = '';
+
+            // Check if fields exist and are not empty
+            if (!empty($spData['fields'])) {
+                $attributeServiceXml .= '<md:AttributeConsumingService index="1">';
+                $attributeServiceXml .= '<md:ServiceName xml:lang="en">ACS</md:ServiceName>';
+
+                foreach ($spData['fields'] as $name => $details) {
+                    $required = isset($details['required']) && $details['required'] ? 'true' : 'false';
```

Comment:
```suggestion
                    $required = empty($details['required']) ? 'false' : 'true';
```
just to simplify the logic, won't this work please ?